pageserver/src/layered_repository/inmemory_layer.rs

```diff
@@ -345,21 +331,21 @@ impl InMemoryLayer {
             self.start_lsn..inner.end_lsn.unwrap(),
         )?;
 
-        let mut do_steps = || -> Result<()> {
-            for (key, vec_map) in inner.index.iter() {
-                // Write all page versions
-                for (lsn, blob_ref) in vec_map.as_slice() {
-                    let mut buf = vec![0u8; blob_ref.size()];
-                    inner.file.read_exact_at(&mut buf, blob_ref.pos())?;
-                    let val = Value::des(&buf)?;
-                    delta_layer_writer.put_value(*key, *lsn, val)?;
-                }
+        let mut buf = Vec::new();
+
+        let mut cursor = inner.file.block_cursor();
+
+        let mut keys: Vec<(&Key, &VecMap<Lsn, u64>)> = inner.index.iter().collect();
+        keys.sort_by_key(|k| k.0);
+
+        for (key, vec_map) in keys.iter() {
+            let key = **key;
+            // Write all page versions
+            for (lsn, pos) in vec_map.as_slice() {
+                cursor.read_blob_into_buf(*pos, &mut buf)?;
+                let val = Value::des(&buf)?;
```

Comment:
When we persisting level (write it as delta layer) we first have to deserialize all value (WAL records) and then once again serialize them to place in delta layer. I wonder if we can avoid this redundant deserialization-serialization and write value just as sequence of bytes?